superset-frontend/src/views/CRUD/data/dataset/AddDataset/DatasetPanel/DatasetPanel.tsx

```diff
@@ -0,0 +1,237 @@
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import { supersetTheme, t, styled } from '@superset-ui/core';
+import Icons from 'src/components/Icons';
+import Table, { ColumnsType, TableSize } from 'src/components/Table';
+import { alphabeticalSort } from 'src/components/Table/sorters';
+// @ts-ignore
+import LOADING_GIF from 'src/assets/images/loading.gif';
+import { ITableColumn } from './types';
+import MessageContent from './MessageContent';
+
+/**
+ * Enum defining CSS position options
+ */
+enum EPosition {
+  ABSOLUTE = 'absolute',
+  RELATIVE = 'relative',
+}
+
+/**
+ * Interface for StyledHeader
+ */
+interface StyledHeaderProps {
+  /**
+   * Determine the CSS positioning type
+   * Vertical centering of loader, No columns screen, and select table screen
+   * gets offset when the header position is relative and needs to be absolute, but table
+   * needs this positioned relative to render correctly
+   */
+  position: EPosition;
+}
+
+const LOADER_WIDTH = 200;
+const SPINNER_WIDTH = 120;
+const HALF = 0.5;
+const MARGIN_MULTIPLIER = 3;
+
+const StyledHeader = styled.div<StyledHeaderProps>`
+  position: ${(props: StyledHeaderProps) => props.position};
+  margin: ${({ theme }) => theme.gridUnit * MARGIN_MULTIPLIER}px;
+  margin-top: ${({ theme }) => theme.gridUnit * (MARGIN_MULTIPLIER + 1)}px;
+  font-size: ${({ theme }) => theme.gridUnit * 6}px;
+  font-weight: ${({ theme }) => theme.typography.weights.medium};
+  padding-bottom: ${({ theme }) => theme.gridUnit * MARGIN_MULTIPLIER}px;
+
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+
+  .anticon:first-of-type {
+    margin-right: ${({ theme }) => theme.gridUnit * (MARGIN_MULTIPLIER + 1)}px;
+  }
+
+  .anticon:nth-of-type(2) {
+    margin-left: ${({ theme }) => theme.gridUnit * (MARGIN_MULTIPLIER + 1)}px;
+  }
+`;
+
+const StyledTitle = styled.div`
+  margin-left: ${({ theme }) => theme.gridUnit * MARGIN_MULTIPLIER}px;
+  margin-bottom: ${({ theme }) => theme.gridUnit * MARGIN_MULTIPLIER}px;
+  font-weight: ${({ theme }) => theme.typography.weights.bold};
+`;
+
+const LoaderContainer = styled.div`
+  padding: ${({ theme }) => theme.gridUnit * 8}px
+    ${({ theme }) => theme.gridUnit * 6}px;
+
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100%;
+`;
+
+const StyledLoader = styled.div`
+  max-width: 50%;
+  width: ${LOADER_WIDTH}px;
+
+  img {
+    width: ${SPINNER_WIDTH}px;
+    margin-left: ${(LOADER_WIDTH - SPINNER_WIDTH) * HALF}px;
+  }
+
+  div {
+    width: 100%;
+    margin-top: ${({ theme }) => theme.gridUnit * MARGIN_MULTIPLIER}px;
+    text-align: center;
+    font-weight: 400;
+    font-size: 16px;
```

Comment:
```suggestion
    font-weight: ${({ theme }) => theme.typography.weights.normal};
    font-size: ${({ theme }) => theme.typography.sizes.l}px;
```
These should use the superset theme.